src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/CreateNextDataSetVersionFunctionTests.cs

```diff
@@ -0,0 +1,499 @@
+            var (dataSet, liveDataSetVersion) = await AddDataSetAndLatestLiveVersion();
+            var (nextReleaseFile, _) = await AddDataAndMetadataFiles(dataSet.PublicationId);
+
+            var durableTaskClientMock = new Mock<DurableTaskClient>(MockBehavior.Strict, "TestClient");
+
+            ProcessDataSetVersionContext? processNextDataSetVersionContext = null;
+            StartOrchestrationOptions? startOrchestrationOptions = null;
+            durableTaskClientMock.Setup(client =>
+                    client.ScheduleNewOrchestrationInstanceAsync(
+                        nameof(ProcessNextDataSetVersionFunction.ProcessNextDataSetVersion),
+                        It.IsAny<ProcessDataSetVersionContext>(),
+                        It.IsAny<StartOrchestrationOptions>(),
+                        It.IsAny<CancellationToken>()))
+                .ReturnsAsync((TaskName _, object _, StartOrchestrationOptions? options, CancellationToken _) =>
+                    options?.InstanceId ?? Guid.NewGuid().ToString())
+                .Callback<TaskName, object, StartOrchestrationOptions?, CancellationToken>(
+                    (_, input, options, _) =>
+                    {
+                        processNextDataSetVersionContext =
+                            Assert.IsAssignableFrom<ProcessDataSetVersionContext>(input);
+                        startOrchestrationOptions = options;
+                    });
+
+            var result = await CreateNextDataSetVersion(
+                dataSetId: dataSet.Id,
+                releaseFileId: nextReleaseFile.Id,
+                durableTaskClientMock.Object);
+
+            VerifyAllMocks(durableTaskClientMock);
+
+            var responseViewModel = result.AssertOkObjectResult<CreateDataSetResponseViewModel>();
+
+            await using var publicDataDbContext = GetDbContext<PublicDataDbContext>();
+
+            // Assert only the original data set exists.
+            var updatedDataSet = Assert.Single(await publicDataDbContext.DataSets
+                .Include(ds => ds.Versions)
+                .ThenInclude(dsv => dsv.Imports)
+                .ToListAsync());
+
+            // Assert that the existing data set is left in its Published state and that its
+            // name, summary and other properties that were created during its first creation
+            // are untouched.
+            Assert.Equal(DataSetStatus.Published, updatedDataSet.Status);
+            Assert.Equal(dataSet.Title, updatedDataSet.Title);
+            Assert.Equal(dataSet.Summary, updatedDataSet.Summary);
+            Assert.Equal(dataSet.PublicationId, updatedDataSet.PublicationId);
+
+            // Assert that the new DataSetVersion is set as the latest draft version, and that the
+            // prior DataSetVersion is still set as the latest live version. 
+            Assert.NotNull(updatedDataSet.LatestDraftVersion);
+            Assert.NotEqual(liveDataSetVersion.Id, updatedDataSet.LatestDraftVersion!.Id);
```

Comment:
would this be better as:
`Assert.Equal(responseViewModel.DataSetVersionId, updatedDataSet.LatestDraftVersion!.Id);`

Reply:
I wanted to test it against the actual entity that was saved, as testing against `responseViewModel.DataSetVersionId` would be a false positive if `responseViewModel.DataSetVersionId` was set incorrectly.